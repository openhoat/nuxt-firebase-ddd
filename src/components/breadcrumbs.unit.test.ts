// @vitest-environment nuxt
import { mountSuspended } from '@nuxt/test-utils/runtime'
import type { VueWrapper } from '@vue/test-utils'
import { beforeAll, beforeEach, describe, expect, test, vi } from 'vitest'
import type { Component } from 'vue'
import { ref } from 'vue'
import type { BreadcrumbsItem } from '~/utils/breadcrumbs-helper'
import type { ComponentWrapper } from '~~/test/utils/test-helper'
import type { RouteLocationNormalizedLoaded } from '#vue-router'

describe('Unit tests', () => {
  describe('components', () => {
    describe('Breadcrumbs', () => {
      let Breadcrumbs: Component
      let wrapper: VueWrapper
      let components: {
        breadcrumbs: ComponentWrapper
      }
      beforeAll(async () => {
        vi.doMock('#app', () => ({
          useRoute: () =>
            ref({
              name: 'bar',
              path: '/foo/bar',
            }) as unknown as RouteLocationNormalizedLoaded,
        }))
        vi.doMock('~/utils/breadcrumbs-helper', () => ({
          buildBreadcrumbsItems: () => {
            const items: BreadcrumbsItem[] = [
              { title: 'Home', to: '/' },
              { title: 'Foo', to: '/foo' },
              { title: 'Bar', to: '/foo/bar' },
            ]
            return items
          },
        }))
        Breadcrumbs = (await import('~/components/Breadcrumbs.vue')).default
        components = {
          get breadcrumbs() {
            return wrapper.findComponent({ name: 'v-breadcrumbs' })
          },
        }
      })
      beforeEach(async () => {
        wrapper = await mountSuspended(Breadcrumbs)
      })
      const expectInitialContent = () => {
        expect(components.breadcrumbs.exists()).toBe(true)
        expect(components.breadcrumbs.isVisible()).toBe(true)
        expect(components.breadcrumbs.html()).toMatchSnapshot()
      }
      test('renders a breadcrumbs given a route', () => {
        // Then
        expectInitialContent()
      })
    })
  })
})
