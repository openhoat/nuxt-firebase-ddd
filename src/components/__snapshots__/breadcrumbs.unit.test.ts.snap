// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Unit tests > components > Breadcrumbs > renders a breadcrumbs given a route 1`] = `
"<ul class="v-breadcrumbs v-breadcrumbs--density-default">
  <li class="v-breadcrumbs__prepend"><i class="mdi-home mdi v-icon notranslate v-theme--dark v-icon--size-default v-icon--start" aria-hidden="true"></i></li>
  <li class="v-breadcrumbs-item"><a class="v-breadcrumbs-item--link">Home</a></li>
  <li class="v-breadcrumbs-divider">/</li>
  <li class="v-breadcrumbs-item"><a class="v-breadcrumbs-item--link">Foo</a></li>
  <li class="v-breadcrumbs-divider">/</li>
  <li class="v-breadcrumbs-item v-breadcrumbs-item--disabled"><a class="v-breadcrumbs-item--link">Bar</a></li>
  <!---->
  <!---->
</ul>"
`;
